source/MaterialXView/Viewer.cpp

```diff
@@ -2084,8 +2130,48 @@ void Viewer::bakeTextures()
     glDrawBuffer(GL_BACK);
 }
 
-void Viewer::draw_contents()
+void Viewer::renderTurnable()
 {
+    unsigned int frameCount = static_cast<unsigned int>(360.0f / _turntableIncrement);
+
+    float currentRotation = _meshRotation[1];
+    _meshRotation[1] = 0.0f;
+
+    mx::FilePath turnableFileName = _captureFilename;
+    const std::string extension = turnableFileName.getExtension();
+    turnableFileName.removeExtension();
+
+    _turntableRotation = 0.0f;
+    for (unsigned int i = 0; i < frameCount; i++)
+    {
+        updateCameras();
+        clear();
+        invalidateShadowMap();
+        renderFrame();
+
+        mx::ImagePtr frameImage = getFrameImage();
+        if (frameImage)
+        {
+            std::stringstream intfmt;
+            intfmt << std::setfill('0') << std::setw(4) << i;
+            std::string saveName = turnableFileName.asString() + "_" + intfmt.str() + "." + extension;
+            if (_imageHandler->saveImage(saveName, frameImage, true))
+            {
+                std::cout << "Wrote turntable frame at angle: " << std::to_string(_turntableRotation) << " to file: " << saveName << std::endl;
+            }
+        }
+
+        _turntableRotation = fmod(_turntableRotation + _turntableIncrement, 360.0f);
+    }
+
+    _meshRotation[1] = currentRotation;
+}
+
+void Viewer::draw_contents()
+{    
+    mx::ScopedTimer frameTime;
+    frameTime.startTimer();
```

Comment:
I don't see where the new frameTime and _frameTime are yet used.  Maybe they were part of an unrelated change, and are not needed in this one?

Reply:
Yes. This pulled over from some frame-rate display code. I'll remove.